app/src/main/java/net/programmierecke/radiodroid2/players/exoplayer/ExoPlayerWrapper.java

```diff
@@ -98,7 +96,47 @@ public void onReceive(Context context, Intent intent) {
             }
         }
     };
+    final class CustomLoadErrorHandlingPolicy extends DefaultLoadErrorHandlingPolicy {
+        final int MIN_RETRY_DELAY_MS = 10;
+        final SharedPreferences sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context);
 
+        int getSanitizedRetryDelaySettingsMs() {
+            return Math.max(sharedPrefs.getInt("settings_retry_delay", 100), MIN_RETRY_DELAY_MS);
```

Comment:
There is a duplication of settings retrieval, maybe we should retrieve them only when playback is started?

Reply:
Actually, I introduced this to be able to change the settings while playing. I guess it's OK to re-read the settings here. It's only done when errors occur and user interventions are likely.

Reply:
Ok, then it would be good idea to add comment and have one way of getting things, e.g. the reading of `settings_retry_delay` is different in two places

Reply:
Right, I guess I'll introduce a local function for this.